packages/react/src/select/Select.tsx

```diff
@@ -0,0 +1,84 @@
+import { useControllableState } from "@radix-ui/react-use-controllable-state";
+import { useSelect, type UseSelectProps } from "downshift";
+import { type ReactNode, useState } from "react";
+
+import { useDelayedState } from "../autocomplete/useDelayedState";
+import { useFieldContext } from "../field-context";
+import { Popover } from "../popover";
+import { SelectContextProvider } from "../select-context";
+
+type SelectProps<Item> = {
+  children?: ReactNode;
+  defaultValue?: Item;
+  disabled?: boolean;
+  onValueChange?: (value: Item) => void;
+  value?: Item;
+} & Pick<
+  UseSelectProps<Item>,
+  | "initialHighlightedIndex"
+  | "isItemDisabled"
+  | "items"
+  | "itemToKey"
+  | "itemToString"
+>;
+
+export function Select<Item>({
+  children,
+  defaultValue,
+  disabled,
+  items,
+  itemToKey = (value) => value,
+  itemToString = (value) => (value ? String(value) : ""),
+  onValueChange,
+  value,
+  ...props
+}: SelectProps<Item>) {
+  const { id: toggleButtonId } = useFieldContext();
+
+  const [selectedItem, setSelectedItem] = useControllableState({
```

Comment:
since we're now controlling the `selectedItem` state we need to pass it down in the context instead of relying on `downshift.selectedItem` which won't have the up-to-date value always

Reply:
it'll be late by 1 render - so on the next render you'll have the correct value - but in the render where the state changes you'll have a stale value